### PR TITLE
CADC-8575 CADC-9186 Updated artifact-discover and artifact-validate r…

### PR DESCRIPTION
**Pull request by yeunga**
…eporting.

Since we now update the harvestskipuri table, we should differentiate between adding and updating the harvestskipuri table. 

Example artifact-discover output:
2021-05-25 14:52:52.228 [Thread-1] INFO  ArtifactHarvester  - ENDDISCOVER: {"total":"120","added":"0","updated":"0","time":"38751","date":"2021-05-25 21:52:13.477"}

Example artifact-validate summary outputs:{"logType":"summary","collection":"IRIS","totalInCAOM":"2580","totalInStorage":"42","totalCorrect":"42","totalDiffChecksum":"0","totalDiffLength":"0","totalDiffType":"0","totalNotInCAOM":"0","totalMissingFromStorage":"2538","totalNotPublic":"0","totalAlreadyInSkipURI":"0","totalNewSkipURI":"0","totalUpdateSkipURI":"2538","time":"30572"}
{"logType":"summary","collection":"IRIS","totalInCAOM":"2580","totalInStorage":"42","totalCorrect":"42","totalDiffChecksum":"0","totalDiffLength":"0","totalDiffType":"0","totalNotInCAOM":"0","totalMissingFromStorage":"2538","totalNotPublic":"0","totalAlreadyInSkipURI":"0","totalNewSkipURI":"2538","totalUpdateSkipURI":"0","time":"30124"}

### CODE REVIEW

File: caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/DownloadArtifactFiles.java [hunk 1]
```diff
@@ -82,6 +82,7 @@
 import ca.nrc.cadc.io.ByteCountInputStream;
 import ca.nrc.cadc.net.HttpDownload;
 import ca.nrc.cadc.net.InputStreamWrapper;
+import ca.nrc.cadc.net.TransientException;
```
Comment (andamian) [inline caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/DownloadArtifactFiles.java:85]: Is this used anywhere?

**Review by andamian — APPROVED** [review]

File: caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/DownloadArtifactFiles.java (hunk 1 above)
Comment (yeunga) [reply caom2-artifact-sync/src/main/java/ca/nrc/cadc/caom2/artifactsync/DownloadArtifactFiles.java]: Good catch. Removed.

**Review by pdowler — CHANGES_REQUESTED** [review]
From the example log messages:
```
"totalAlreadyInSkipURI":"0","totalNewSkipURI":"0","totalUpdateSkipURI":"2538"
"totalAlreadyInSkipURI":"0","totalNewSkipURI":"2538","totalUpdateSkipURI":"0"
```
From the first line, weren't the updated records already in the skip table? So totalAlreadyInSkipURI is clearly not a count of total skip records before or after... it's really a count of the subset of existing skip records that were looked at but not modified? That's pretty subtle and will be confusing.

In caom2harvester --validate mode, the summary report has 4 numbers (counts):
```
total + " mismatches: " + mismatch + " known: " + known + " new: " + added;
```
(total is number of observations examined, less interesting). `mismatches` is the total number of skip records. This mode does update skip records, but it doesn't distinguish that it updated the (already) known records. It should always be true that `mismatch == known + added` or it is confusing (sign of bug). When I look at that and I look at totalAlreadyInSkipURI the latter isn't very clear to me: it sounds like it is the total number of skip records but it isn't. You could report skips,updated,new where skips == updated + new... the label style is fine but the meanings have to be clear and some fields adding up to another always helps people understand.

**Comment by yeunga** [comment]
artifact-validate summary now looks this:{"logType":"summary","collection":"IRIS","totalInCAOM":"2580","totalInStorage":"16","totalCorrect":"0","totalDiffChecksum":"0","totalDiffLength":"0","totalDiffType":"0","totalNotInCAOM":"16","totalMissingFromStorage":"2580","totalNotPublic":"0","totalInSkipURI":"2580","totalNewSkipURI":"0","totalUpdateSkipURI":"2580","time":"28364"}

**Review by pdowler — APPROVED** [review]

**Pull request merged**